packages/dui3/store/hostApp.ts

```diff
@@ -0,0 +1,416 @@
+    model.progress = undefined
+  }
+
+  app.$sendBinding?.on('setModelSendResult', setModelSendResult)
+
+  /// RECEIVE STUFF
+  const receiveModel = async (modelCardId: string) => {
+    const model = documentModelStore.value.models.find(
+      (m) => m.modelCardId === modelCardId
+    ) as IReceiverModelCard
+
+    void trackEvent(
+      'DUI3 Action',
+      { name: 'Receive', expired: model.expired },
+      model.accountId
+    )
+
+    model.report = undefined
+    model.error = undefined
+    model.displayReceiveComplete = false
+    model.hasDismissedUpdateWarning = true
+    model.progress = { status: 'Starting to receive...' }
+    await app.$receiveBinding.receive(modelCardId)
+  }
+
+  const receiveModelCancel = async (modelCardId: string) => {
+    const model = documentModelStore.value.models.find(
+      (m) => m.modelCardId === modelCardId
+    ) as IReceiverModelCard
+    await app.$receiveBinding.cancelReceive(modelCardId)
+    void trackEvent('DUI3 Action', { name: 'Receive Cancel' }, model.accountId)
+    model.progress = undefined
+  }
+
+  const setModelReceiveResult = async (args: {
+    modelCardId: string
+    bakedObjectIds: string[]
+    conversionResults: ConversionResult[]
+  }) => {
+    const model = documentModelStore.value.models.find(
+      (m) => m.modelCardId === args.modelCardId
+    ) as IReceiverModelCard
+
+    model.progress = undefined
+    model.displayReceiveComplete = true
+    model.bakedObjectIds = args.bakedObjectIds
+    model.report = args.conversionResults
+
+    // NOTE: going through this method to ensure state sync between FE and BE. It's because of a very weird rhino bug on first receives, ask dim and he will cry
+    // TODO: check if it's still needed - we can store the bakedobject ids straigth into the receive ops in .net. Is the above reproducible?
+    await patchModel(model.modelCardId, {
+      bakedObjectIds: args.bakedObjectIds
+    })
+  }
+
+  app.$receiveBinding?.on('setModelReceiveResult', setModelReceiveResult)
+
+  // GENERIC STUFF
+  const handleModelProgressEvents = (args: {
+    modelCardId: string
+    progress?: ModelCardProgress
+  }) => {
+    const model = documentModelStore.value.models.find(
+      (m) => m.modelCardId === args.modelCardId
+    ) as IModelCard
+    model.progress = args.progress
+  }
+
+  const handleModelError = (args: { modelCardId: string; error: string }) => {
+    const model = documentModelStore.value.models.find(
+      (m) => m.modelCardId === args.modelCardId
+    ) as IModelCard
+    model.progress = undefined
+    model.error = args.error
+  }
+
+  // NOTE: all bindings that need to send these model events should register.
+  // EG, new binding "mapper binding" wants to send errors to the model card should
+  // be registed here. Why? Each binding gets its own "bridge" parent in .NET, which
+  // is hoisted as a separate global js object.
+  app.$sendBinding?.on('setModelProgress', handleModelProgressEvents)
```

Comment:
None of the event handlers anywhere I've seen are cleaned up, which can potentially cause memory leaks